File: maratona2016.1/src/br/ufs/uri/uri1710.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package br.ufs.uri;

import java.util.ArrayList;
import java.util.Scanner;

public class uri1710 {
    public static void main() {
        Scanner s = new Scanner(System.in);
        
        int m = s.nextInt();
        int n = s.nextInt();
        
        int max = 0;
        
        ArrayList<Plano> planos = new ArrayList<Plano>();
        
        int i;
        for(i = 0; i < m; i++){
            int a = s.nextInt();
            int b = s.nextInt();
            int c = s.nextInt();
            int d = s.nextInt();
            planos.add(new Plano(a,b,c,d));
        }
        Plano maiorPlano = null;
        for(i = 0; i < n; i++){
            int j;
            int x = s.nextInt();
            int y = s.nextInt();
            int z = s.nextInt();
            for(j = 0; j < m; j++){
                Plano p = planos.get(i);
                if(p.isInPlano(x, y, z)){
                    if(maiorPlano == null || maiorPlano.mundos < p.mundos) {
                        maiorPlano = p;
                    }
                    break;
                }
            }
        }
        
        System.out.println(maiorPlano.mundos);
    }
}


class Plano{
    public int a, b, c, d;
    public int mundos = 0;

    Plano(int a, int b, int c, int d){
        this.a = a;
        this.b = b;
        this.c = c;
        this.d = d;
    }
    
    public boolean isInPlano(int x, int y, int z){
        if(x*a + y*b + z*c == d){
            mundos++;
            return true;
        }
        return false;
    }
}

File: uri/u1710/uri1710.cpp
#include <stdio.h>

using namespace std;

struct Plano {
    int a=0, b=0, c=0, d=0;
    int mundos = 0;
};

int isInPlano(Plano p, int x, int y, int z) {
    int pos = ((x * p.a) + (y * p.b) + (z * p.c));
    if (pos == p.d) {
        p.mundos++;
        return 1;
    }
    return 0;
}

int main() {
    int m, n;
    scanf("%d %d", &m, &n);
    Plano planos[m];

    int i;
    for (i = 0; i < m; i++) {
        int a, b, c, d;
        scanf("%d %d %d %d", &a, &b, &c, &d);
        planos[i].a = a;
        planos[i].b = b;
        planos[i].c = c;
        planos[i].d = d;
    }

    Plano maiorPlano;
    for (i = 0; i < n; i++) {
        if (i == 0) maiorPlano = planos[0];
        int j, x, y, z;
        scanf("%d %d %d", &x, &y, &z);

        for (j = 0; j < m; j++) {
            if (isInPlano(planos[j], x, y, z)) {
                if (maiorPlano.mundos < planos[j].mundos) {
                    maiorPlano = planos[j];
                }
                break;
            }
        }
    }

    printf("%d\n", maiorPlano.mundos);
}


